src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+from zenml.logger import get_logger
+from zenml.model_registries.base_model_registry import (
+    BaseModelRegistry,
+    ModelRegistration,
+    ModelVersion,
+    ModelVersionStage,
+)
+from zenml.stack.stack import Stack
+from zenml.stack.stack_validator import StackValidator
+
+logger = get_logger(__name__)
+
+
+class MLFlowModelRegistry(BaseModelRegistry):
+    """Register models using MLflow."""
+
+    _client: Optional[MlflowClient] = None
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the model registry and validate the tracking uri.
+
+        Args:
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+        super().__init__(*args, **kwargs)
+
+    @property
+    def config(self) -> MLFlowModelRegistryConfig:
+        """Returns the `MLFlowModelRegistryConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(MLFlowModelRegistryConfig, self._config)
+
+    def configure_mlflow(self) -> None:
+        """Configures the MLflow Client with the experiment tracker config."""
+        experiment_tracker = Client().active_stack.experiment_tracker
+        if not isinstance(experiment_tracker, MLFlowExperimentTracker):
+            raise get_missing_mlflow_experiment_tracker_error()
+        experiment_tracker.configure_mlflow()
+
+    @property
+    def mlflow_client(self) -> MlflowClient:
+        """Get the MLflow client.
+
+        Raises:
+            get_missing_mlflow_experiment_tracker_error: If the stack does not
+                contain an MLFlow experiment tracker.
+
+        Returns:
+            The MLFlowClient.
+        """
+        if not self._client:
+            self.configure_mlflow()
+            self._client = mlflow.tracking.MlflowClient()
+        return self._client
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Validates that the stack contains an mlflow experiment tracker.
+
+        Returns:
+            A StackValidator instance.
+        """
+
+        def _validate_stack_requirements(stack: "Stack") -> Tuple[bool, str]:
+            """Validates that the experiment tracker is an mlflow experiment tracker.
+
+            Args:
+                stack: The stack to validate.
+
+            Returns:
+                A tuple of (is_valid, error_message).
+            """
+            # Validate that the experiment tracker is an mlflow experiment tracker.
+            experiment_tracker = stack.experiment_tracker
+            assert experiment_tracker is not None
+            if experiment_tracker.flavor != "mlflow":
+                return False, (
+                    "The MLflow model registry requires a MLflow experiment "
+                    "tracker. You should register a MLflow experiment "
+                    "tracker to the stack using the following command: "
+                    "`zenml stack register experiment_tracker ..."
```

Comment:
This registers a stack called experiment tracker, probably something like this instead? `zenml experiment-tracker register <NAME> --flavor=mlflow`